addons/stock/models/stock_picking.py

```diff
@@ -417,17 +418,28 @@ def _compute_state(self):
         - Done: if the picking is done.
         - Cancelled: if the picking is cancelled
         '''
+        picking_moves_state_map = defaultdict(dict)
+        picking_move_lines = defaultdict(set)
+        for move in self.env['stock.move'].search([('picking_id', 'in', self.ids)]):
+            picking_id = move.picking_id
+            move_state = move.state
+            picking_moves_state_map[picking_id.id].update({
+                'any_draft': picking_moves_state_map[picking_id.id].get('any_draft', False) or move_state == 'draft',
+                'all_cancel': picking_moves_state_map[picking_id.id].get('all_cancel', True) and move_state == 'cancel',
+                'all_cancel_done': picking_moves_state_map[picking_id.id].get('all_cancel_done', True) and move.state in ['cancel', 'done'],
+            })
```

Comment:
instead:

```python
            picking_moves_state_map[picking_id.id].update(
                any_draft=picking_moves_state_map[picking_id.id].get('any_draft', False) or move_state == 'draft',
                all_cancel=picking_moves_state_map[picking_id.id].get('all_cancel', True) and move_state == 'cancel',
                all_cancel_done=picking_moves_state_map[picking_id.id].get('all_cancel_done', True) and move_state in ('cancel', 'done'),
            )
```
(`move_state` instead of `move.state`, avoid creation of new dict, `in` with tuple.)

Reply:
Oh yes nicely spotted 😮